### PR TITLE
Feature/df 243 domain add postaddress

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import javax.persistence.Access;
+import javax.persistence.AccessType;
+import javax.persistence.Embeddable;
+import java.util.Objects;
+
+@Embeddable
+@Access(AccessType.FIELD)
+public class Address {
+
+    // TODO @Size(max = 255)
```
Comment (birom) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java]: This field is required.

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    private String name;
+
+    // TODO @Size(max = 60)
+    private String street;
+
+    // TODO @Size(max = 10)
+    private String houseNumber;
+
+    // TODO @Size(max = 10)
```
Comment (birom) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java]: This field is required.

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    private String postalCode;
+
+    // TODO @Size(max = 100)
```
Comment (birom) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java]: This field is required.

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    // TODO @Size(max = 20)
+    private String postOfficeBoxNumber;
+
+    // TODO @Size(max = 10)
+    private String postOfficeBoxPostalCode;
+
+    // TODO @Size(max = 100)
+    private String postOfficeBoxCity;
+
+    // TODO @Size(max = 2)
```
Comment (birom) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java]: This field is required, and must be a vaid countryIsoCode.

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Company.java [hunk 5]
```diff
@@ -47,63 +29,64 @@
...
-        this.postalCode = builder.postalCode;
-        this.city = builder.city;
-        //this.countryIsoCode = Condition.notBlank(builder.countryIsoCode, "Country of a company can't be null");
-        //todo: Review the countryIsoCode, because most of the jobs from x28 do not have country.
-        this.countryIsoCode = builder.countryIsoCode;
-        this.postOfficeBoxNumber = builder.postOfficeBoxNumber;
-        this.postOfficeBoxPostalCode = builder.postOfficeBoxPostalCode;
-        this.postOfficeBoxCity = builder.postOfficeBoxCity;
+        this.address = new Address.Builder()
+                .setName(Condition.notBlank(builder.name, "Name of a company can't be null"))
```
Comment (birom) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Company.java]: I think the address and its builder must be responsibe for the validation.

Comment (cjpsycos) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Company.java]: Good question.
Is an address really responsible for validation in any use-case?

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java (hunk 1 above)
Comment (cjpsycos) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java]: this was not in the company

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java (hunk 2 above)
Comment (cjpsycos) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java]: this was not in the company

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java (hunk 3 above)
Comment (cjpsycos) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Address.java]: this was not on the company

**Pull request merged**